build-compiler-5.0.1.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
```

Comment:
`compiler` is too generic a name... `clang`, probably?

Reply:
`clang` is definitely better than `compiler`, but thinking about it now I might actually suggest to call it `clang-llvm`, or even `clang+llvm`, because the LLVM version (`5.0.1`) is what's used in the release download url from the LLVM downloads page, *not* the clang version. Also, the MacOS binary release download unpacks into an archive named `clang+llvm-5.0.1-<platform>.tar.xz`, so calling it `clang+llvm` would match that convention. It could be `clang-llvm` too, but I prefer `clang+llvm` because that makes it clear which part of `clang+llvm-5.0.1.tar.gz` is the package name and which part is the version.

Reply:
Are you planning to package something in here other than `clang` itself?

Reply:
Yes, the C and C++ standard library headers. The C standard library headers are already included -- I hadn't tested the C++ standard library headers because we don't have a `python_dist()` target with C++ code in `testprojects/` yet in the pants repo. These would be provided with the compiler so that they are compatible with the compiler, and so we don't have to rely on headers existing in standard locations (which is a requirement for an internal consumer of `python_dist()` with native code).

I'm not planning to package anything that's not considered part of clang, however. It might make sense to just call it `clang`, by that logic. Since we're not packaging any other LLVM tools, it might be confusing/surprising to call it `clang+llvm`.

After concentrating on this for a bit, my vote goes to just calling it `clang`.

Reply:
Changed to `build-clang-5.0.1.sh`, producing `clang.tar.gz` in supportdir `build-support/bin/clang` in ad67ff4.